### PR TITLE
chore: add more to/from le/be bits/bytes edge case tests

### PR DESCRIPTION
**Pull request by michaeljklein**
# Description

## Summary\*

Adding tests for:
- `from_be_bytes`, `to_be_bytes`
- `from_le_bytes`, `to_le_bytes`
- `from_be_bits`, `to_be_bits`
- `from_le_bits`, `to_le_bits`

involving:
- `modulus_be_bits`
-  `modulus_le_bits`
-  `modulus_be_bytes`
-  `modulus_le_bytes`

## Documentation\*

Check one:
- [x] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: noir_stdlib/src/field/mod.nr [hunk 1]
```diff
@@ -573,4 +575,174 @@ mod tests {
...
+
+            let p_bytes: [u8; 32] = p.to_le_bytes();
+            assert_eq(p_bytes, [0; 32]);
+            assert(p_bytes.as_slice() != modulus_le_bytes());
+        }
+    }
+
+    /// Convert a little endian bit array to a field element.
+    /// If the provided bit array overflows the field modulus then the Field will silently wrap around.
+    pub fn from_le_bits<let N: u32>(bits: [u1; N]) -> Field {
```
Comment (asterite) [inline noir_stdlib/src/field/mod.nr]: Does this need to be `pub`?

File: noir_stdlib/src/field/mod.nr [hunk 2]
```diff
@@ -573,4 +575,174 @@ mod tests {
...
+        for i in 0..N {
+            result += (bits[i] as Field) * v;
+            v = v * 2;
+        }
+        result
+    }
+
+    /// Convert a big endian bit array to a field element.
+    /// If the provided bit array overflows the field modulus then the Field will silently wrap around.
+    pub fn from_be_bits<let N: u32>(bits: [u1; N]) -> Field {
```
Comment (asterite) [inline noir_stdlib/src/field/mod.nr]: Does this need to be `pub`?

File: noir_stdlib/src/field/mod.nr [hunk 3]
```diff
@@ -573,4 +575,174 @@ mod tests {
...
+
+            assert(q_bytes != p_plus_1_bytes);
+
+            assert_eq(modulus_le_bytes().len(), 32);
+            let p = Field::from_le_bytes::<32>(modulus_le_bytes().as_array());
+            assert_eq(p, 0);
+
+            let p_bytes: [u8; 32] = p.to_le_bytes();
+            assert_eq(p_bytes, [0; 32]);
+            assert(p_bytes.as_slice() != modulus_le_bytes());
```
Comment (asterite) [inline noir_stdlib/src/field/mod.nr]: It might be good to comment why this equality doesn't hold. At first glance I see `p` comes from `modules_le_bytes`, then `p` is converted back to bytes using `to_le_bytes` so I don't know why they aren't equal.

It might also be good to put comments on each chunk of this test explaining what we are testing as it seems multiple things are tested. I can kind of guess based on the var names, but comments would make it a bit clearer, maybe.
